pkg/gadgets/socket-collector/gadget.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package socketcollector
+
+import (
+	"fmt"
+	"sync"
+
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/types"
+
+	gadgetv1alpha1 "github.com/kinvolk/inspektor-gadget/pkg/api/v1alpha1"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/socket-collector/tracer"
+)
+
+type Trace struct {
+	resolver gadgets.Resolver
+}
+
+type TraceFactory struct {
+	gadgets.BaseFactory
+	mu     sync.Mutex
+	traces map[string]*Trace
+}
+
+func setStatusError(trace *gadgetv1alpha1.Trace, err string) {
+	trace.Status.OperationError = err
+	trace.Status.Output = ""
+	trace.Status.State = "Stopped"
+}
```

Comment:
I wonder if this could be a helper in pkg/gadgets/helpers.go

Reply:
I think we should use it in general because in this way we ensure `State` and `Output` are updated accordingly if user starts the gadget more than once. For instance in a situation where the first time it successes but the second one it doesn't, we don't want to have an `OperationError` not null and a valid output (From the previous execution).